src/cim/components/ClusterDeployment/ClusterDeploymentHostDiscoveryTable.tsx

```diff
@@ -50,12 +52,15 @@ const ClusterDeploymentHostDiscoveryTable: React.FC<ClusterDeploymentHostDiscove
   const [isMassChangeHostOpen, setMassChangeHostOpen] = React.useState(false);
   const [isMassApproveOpen, setMassApproveOpen] = React.useState(false);
   const [selectedHostIDs, setSelectedHostIDs] = React.useState<string[]>([]);
+  const { setFieldValue } = useFormikContext<ClusterDeploymentHostsDiscoveryValues>();
+
   const onSelect = (host: Host, isSelected: boolean) => {
     if (isSelected) {
       setSelectedHostIDs([...selectedHostIDs, host.id]);
     } else {
       setSelectedHostIDs(selectedHostIDs.filter((sa) => sa !== host.id));
     }
+    setFieldValue('selectedHostIds', [...selectedHostIDs, host.id]);
```

Comment:
this needs to consider `isSelected` prop.
isSelected === true => add to `selectedHostIds`
isSelected === false => remove from `selectedHostIds`

Reply:
done!